### PR TITLE
Schema and method name improvements

### PR DESCRIPTION
**Pull request by hcsa73**
Standardizes names across service modules, and improves them based on user feedback

### CODE REVIEW

File: services/argus/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,73 @@
+## v0.8.0 (2023-12-19)
+
+API methods and structs were renamed to have the same look and feel and according to user feedback.
```
Comment (joaopalet) [inline services/argus/CHANGELOG.md:3]:
```suggestion
API methods and structs were renamed to have the same look and feel across all STACKIT services and according to user feedback.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,135 @@
+## Release (2023-12-19)
+
+API methods and structs were renamed to have the same look and feel and according to user feedback.
```
Comment (joaopalet) [inline CHANGELOG.md:3]:
```suggestion
API methods and structs were renamed to have the same look and feel across all services and according to user feedback.
```

Comment (joaopalet) [reply CHANGELOG.md:3]: Same for the service changelogs

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,135 @@
...
+
```
Comment (joaopalet) [inline CHANGELOG.md:4]: Should we also mention here the main ideas that drove the changes? E.g. `Get<something plural>` to `List...`, `Update...` and `PartialUpdate...`

Comment (hcsa73) [reply CHANGELOG.md:11]: Added summary, please check

File: services/dns/wait/wait.go [hunk 4]
```diff
@@ -46,8 +46,8 @@ func CreateZoneWaitHandler(ctx context.Context, a APIClientInterface, projectId,
 	return handler
 }
 
-// UpdateZoneWaitHandler will wait for zone update
-func UpdateZoneWaitHandler(ctx context.Context, a APIClientInterface, projectId, instanceId string) *wait.AsyncActionHandler[dns.ZoneResponse] {
+// PartialUpdateZoneWaitHandler will wait for zone update
```
Comment (joaopalet) [inline services/dns/wait/wait.go:49]: Is is relevant for the waiter if the update was partial or not? If there is is both a `PartialUpdate...` and `Update...`, we would only need one waiter for both updates, right?

Comment (hcsa73) [reply services/dns/wait/wait.go:49]: It's so it matches the corresponding method.

Added it to the changelogs and created a `PartialUpdateInstanceWaitHandler` for mongodbflex (which has both `Update` and `PartialUpdate`), which I think is the only case where we have waiters for update operations that can be both partial and full. Please check

**Review by joaopalet — APPROVED** [review]

**Pull request merged**